jwst/source_catalog/source_catalog_step.py

```diff
@@ -46,17 +47,14 @@ def process(self, input):
 
             self.log.info('Detected {0} sources'.format(len(catalog)))
 
-            old_suffixes = ['i2d']
-            output_dir = self.search_attr('output_dir')
-            cat_filepath = replace_suffix_ext(model.meta.filename,
-                                              old_suffixes, 'cat',
-                                              output_ext='ecsv',
-                                              output_dir=output_dir)
-            catalog.write(cat_filepath, format='ascii.ecsv', overwrite=True)
-            self.log.info('Wrote source catalog: {0}'
-                          .format(cat_filepath))
-            model.meta.source_catalog.filename = cat_filepath
+            if self.save_results:
+                cat_filepath = self.make_output_path()
```

Comment:
@stscieisenhamer  Does `make_output_path()` include the filename or just the path?  If it's the latter, then this is the bug.  Before this change, `cat_filepath` was both the full path+filename.  The catalog filename is based on `model.meta.filename`.

CC: @sosey

Reply:
when I update the configspec, this is the filename I get:
```
  Use SourceCatalog.to_table() instead.
  catalog = photutils.properties_table(source_props, columns=columns)

2018-06-08 13:06:52,249 - stpipe.SourceCatalogStep - INFO - Detected 49 sources
2018-06-08 13:06:52,294 - stpipe.SourceCatalogStep - INFO - Wrote source catalog: step_SourceCatalogStep_cat.ecsv
```

Reply:
Hmm, that's not what we want.